### PR TITLE
add ability to clear chat history

### LINKED ISSUE #1336: Clearing the chat
Ref https://github.com/nextcloud/spreed/issues/1243

The issue comes with 2 things that need implementing:
1. A button to clear the chat from the mobile app
2. The system message that the chat was cleared needs to be understood and delete all locally saved and displayed messages

### PR DESCRIPTION
**Pull request by mahibi**
- moderators can clear the chat in the conversations settings
- chat is cleared by system message CLEARED_CHAT

Fix #1336

### CODE REVIEW

**Comment by mahibi** [comment]
will be rebased etc but must leave for kindergarten now...

**Comment by splitt3r** [comment]
> 
> 
> will be rebased etc but must leave for kindergarten now...

first things first 👍🏻

File: app/src/main/java/com/nextcloud/talk/utils/ApiUtils.java [hunk 1]
```diff
@@ -394,4 +394,8 @@ public static String getUrlForUserFields(String baseUrl) {
     public static String getUrlToSendLocation(String baseUrl, String roomToken) {
         return baseUrl + ocsApiVersion + "/apps/spreed/api/v1/chat/" + roomToken + "/share";
     }
+
+    public static String getUrlToClearChatHistory(String baseUrl, String roomToken) {
+        return baseUrl + ocsApiVersion + "/apps/spreed/api/v1/chat/" + roomToken;
```
Comment (nickvergessen) [inline app/src/main/java/com/nextcloud/talk/utils/ApiUtils.java:399]: Hardcoded api version. Please inject it like woth all the other api calls

File: app/src/main/java/com/nextcloud/talk/controllers/ConversationInfoController.kt [hunk 2]
```diff
@@ -488,6 +490,33 @@ class ConversationInfoController(args: Bundle) :
         }
     }
 
+    private fun clearHistory() {
+        ncApi?.clearChatHistory(
+            credentials,
+            ApiUtils.getUrlToClearChatHistory(conversationUser!!.baseUrl, conversationToken)
```
Comment (nickvergessen) [inline app/src/main/java/com/nextcloud/talk/controllers/ConversationInfoController.kt:496]: ApiUtils.getChatApiVersion() is missing before this and should be injected

File: app/src/main/java/com/nextcloud/talk/controllers/ConversationInfoController.kt [hunk 3]
```diff
@@ -529,8 +558,10 @@ class ConversationInfoController(args: Bundle) :
 
                         if (conversationCopy!!.canModerate(conversationUser)) {
                             binding.addParticipantsAction.visibility = View.VISIBLE
+                            binding.clearConversationHistory.visibility = View.VISIBLE
```
Comment (nickvergessen) [inline app/src/main/java/com/nextcloud/talk/controllers/ConversationInfoController.kt:561]: Needs to be wrapped by a capability check for `clear-history`

Ref https://github.com/nextcloud/spreed/pull/6023

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/nextcloud/talk/controllers/ConversationInfoController.kt [hunk 4]
```diff
@@ -529,8 +561,12 @@ class ConversationInfoController(args: Bundle) :
 
                         if (conversationCopy!!.canModerate(conversationUser)) {
                             binding.addParticipantsAction.visibility = View.VISIBLE
+                            if (CapabilitiesUtil.hasSpreedFeatureCapability(conversationUser, "clear-history")) {
+                                binding.clearConversationHistory.visibility = View.VISIBLE
+                            }
```
Comment (nickvergessen) [inline app/src/main/java/com/nextcloud/talk/controllers/ConversationInfoController.kt:566]:
```suggestion
                            } else {
                                binding.clearConversationHistory.visibility = View.GONE
                            }
```

**Review by nickvergessen — APPROVED** [review]
Works

**Comment by nickvergessen** [comment]
@timkrueger are you able to test things like this? Or need a crash course?

**Review by AndyScherzinger — APPROVED** [review]

**Pull request merged**